File: go-sdk/pkg/models/operations/deletelegalholdpolicyassignmentsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteLegalHoldPolicyAssignmentsIDPathParams struct {
	LegalHoldPolicyAssignmentID string `pathParam:"style=simple,explode=false,name=legal_hold_policy_assignment_id"`
}

type DeleteLegalHoldPolicyAssignmentsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteLegalHoldPolicyAssignmentsIDPathParams
}

type DeleteLegalHoldPolicyAssignmentsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/deletelegalholdpolicyassignmentsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteLegalHoldPolicyAssignmentsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=legal_hold_policy_assignment_id" })
  legalHoldPolicyAssignmentId: string;
}


export class DeleteLegalHoldPolicyAssignmentsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteLegalHoldPolicyAssignmentsIdPathParams;
}


export class DeleteLegalHoldPolicyAssignmentsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
